metadata/pkg/validator/generalQueryValidator.go

```diff
@@ -0,0 +1,48 @@
+package validator
+
+import (
+	"github.com/opensds/multi-cloud/metadata/pkg/constants"
+	pb "github.com/opensds/multi-cloud/metadata/proto"
+)
+
+type GeneralQueryValidator struct {
+}
+
+func (v *GeneralQueryValidator) isValidQuery(in *pb.ListMetadataRequest) (bool, error) {
+	// TODO: validate backend name if present is valid according to backend naming conventions
+
+	// validate size of object and operator is valid
+	okie, err := isSizeParamsValid(in.SizeOfBucketInBytes, in.BucketSizeOperator)
+	if !okie {
+		return okie, err
+	}
+
+	// validate size of object and operator is valid
+	okie, err = isSizeParamsValid(in.SizeOfObjectInBytes, in.ObjectSizeOperator)
+
+	return okie, err
+}
+
+func isSizeParamsValid(sizeInBytes int64, operator string) (bool, error) {
+	switch operator {
+	case constants.EQUAL_OPERATOR:
+
+	case constants.LESS_THAN_EQUAL_OPERATOR:
+
+	case constants.LESS_THAN_OPERATOR:
+
+	case constants.GREATER_THAN_EQUAL_OPERATOR:
+
+	case constants.GREATER_THAN_OPERATOR:
+		// operator is valid
+		break
```

Comment:
why do we need break here?

Reply:
moved all validation logic to validator file